### PR TITLE
Round 4 of JUnit test refactorings

### PR DESCRIPTION
**Pull request by ghale**
This refactors the last of the tests using the old `JUnitMultiVersionTest` and removes its infrastructure.

### CODE REVIEW

**Comment by ghale** [comment]
@bot-gradle test this

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/junitplatform/JUnitPlatformTestRewriter.groovy
Comment (octylFractal) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/junitplatform/JUnitPlatformTestRewriter.groovy:1]: :tada:

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testng/TestNGMultiVersionTest.groovy [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @Override
+    AbstractJUnitMultiVersionIntegrationTest.TestSourceConfiguration getTestSourceConfiguration() {
+        return new TestNGTestSourceConfiguration()
+    }
+
+    static class TestNGBuildSourceConfiguration implements AbstractJUnitMultiVersionIntegrationTest.BuildScriptConfiguration {
+        @Override
+        String getTestFrameworkDependencies(String sourceSet) {
+            return """
+                    testImplementation 'org.testng:testng:${MultiVersionIntegrationSpec.version}'
```
Comment (octylFractal) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testng/TestNGMultiVersionTest.groovy:37]: Shouldn't this be handling the `sourceSet`?

**Review by octylFractal — COMMENTED** [review]
Will there be a round 5 PR to refactor tests using `JUNIT_VINTAGE_JUPITER`?

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testng/TestNGMultiVersionTest.groovy (hunk 1 above)
Comment (ghale) [reply subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testng/TestNGMultiVersionTest.groovy]: Yep, good catch.  Fixed.

**Comment by ghale** [comment]
> Will there be a round 5 PR to refactor tests using JUNIT_VINTAGE_JUPITER?

Yes, I've got another PR ready that reorganizes a bunch of this stuff and refactors one of these tests.  Then I have another set of changes staged that adjusts some of the coverage versions, groups, etc and as part of it refactors the other tests that use JUNIT_VINTAGE_JUPITER.

**Comment by ghale** [comment]
@bot-gradle test and merge

**Comment by ghale** [comment]
@bot-gradle test and merge

**Pull request merged**